src/main/java/com/ericsson/gerrit/plugins/eiffel/messaging/EiffelEventSender.java

```diff
@@ -89,12 +106,23 @@ public EiffelEventSender setEiffelEventType(final String eiffelType) {
         return this;
     }
 
-    private void generateAndPublish()
+    private String generateAndPublish()
             throws URISyntaxException, IOException, MissingConfigurationException,
             HttpRequestFailedException {
         assembleRequest();
         final ResponseEntity response = httpRequest.performRequest();
         verifyResponse(response);
+
+        String generatedEventId = getGeneratedEventId(response);
+        return generatedEventId;
+    }
+
+    private String getGeneratedEventId(final ResponseEntity response) {
+        final String jsonBody = response.getBody();
+        final JsonObject responseBody = new JsonParser().parse(jsonBody).getAsJsonObject();
+        final JsonArray events = responseBody.get("events").getAsJsonArray();
+        final String generatedEventId = events.get(0).getAsJsonObject().get("id").getAsString();
```

Comment:
Are you certain that you always have a "0" or an "id"? This could cause a null pointer exception.

Reply:
As long as we are getting 200 it will be this structure. If we get something else, we should maybe throw an exception and never get to this place.

Reply:
I think it may be fine the way it is. verifyResponse will check the status code and return an exception if it is 500 or 503. This is to make sure it retries though so for any other status code no exception is thrown but I guess this could be expanded in there somehow.